File: app-restaurant/src/main/java/com/restaurant/apprestaurant/model/MenuItem.java
package com.restaurant.apprestaurant.model;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;

@Entity
@Getter
@Setter
@EqualsAndHashCode
@ToString
public class MenuItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private double price;

    @ManyToOne
    private Category category;

    public MenuItem(){}

    public MenuItem(String name,double price, Category category){
        this.name = name;
        this.price = price;
        this.category = category;
    }

    public MenuItem(Long id, String name,double price, Category category){
        this.id = id;
        this.name = name;
        this.price = price;
        this.category = category;
    }





}

File: app-restaurant-client/src/app/MenuItem.ts
import { Category } from './category';

export class MenuItem implements MenuItemInterface {
  public id: number;
  public name: string;
  public price: number;
  public category: Category;

  constructor(miCfg: MenuItemInterface) {
    this.id = miCfg.id;
    this.name = miCfg.name;
    this.price = miCfg.price;
    this.category = miCfg.category;
  }
}

export interface MenuItemInterface {
  id: number;
  name: string;
  price: number;
  category: Category;
}
